module/VuFind/src/VuFind/Controller/NotificationsController.php

```diff
@@ -0,0 +1,433 @@
+<?php
+
+/**
+ * Notifications Controller
+ *
+ * PHP version 8
+ *
+ * Copyright (C) effective WEBWORK GmbH 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Johannes Schultze <schultze@effective-webwork.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Controller;
+
+use ElGigi\CommonMarkEmoji\EmojiExtension;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\Stdlib\ArrayObject;
+use League\CommonMark\Environment\Environment;
+use League\CommonMark\Extension\Autolink\AutolinkExtension;
+use League\CommonMark\Extension\CommonMark\CommonMarkCoreExtension;
+use League\CommonMark\MarkdownConverter;
+use VuFind\Form\BroadcastsForm;
+use VuFind\Form\PagesForm;
+
+/**
+ * Controls the configuration and display of notifications
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Johannes Schultze <schultze@effective-webwork.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class NotificationsController extends \VuFind\Controller\AbstractBase
+{
+    /**
+     * Notifications config
+     *
+     * @var array
+     */
+    private $config;
```

Comment:
We try to avoid private properties in VuFind code to allow greater extensibility; I'd recommend changing this to protected.

Reply:
Is changed to protected.